packages/ckeditor5-dev-dependency-checker/lib/checkversionmatch.js

```diff
@@ -149,16 +155,23 @@ function checkDependenciesMatch( { packageJsons, devDependenciesFilter, expected
  * @param {object} options.versionExceptions
  * @param {object} options.versionsCache
  * @param {boolean} options.allowRanges
+ * @param {boolean} options.useWorkspace
  * @return {object.<string, string>} expectedDependencies
  */
-function getExpectedDepsVersions( { packageJsons, devDependenciesFilter, versionExceptions, versionsCache, allowRanges } ) {
+function getExpectedDepsVersions( { packageJsons, devDependenciesFilter, versionExceptions, versionsCache, allowRanges, useWorkspace } ) {
 	return packageJsons.reduce( ( expectedDependencies, packageJson ) => {
 		DEPENDENCY_TYPES.forEach( dependencyType => {
 			if ( !packageJson[ dependencyType ] ) {
 				return;
 			}
 
 			Object.entries( packageJson[ dependencyType ] ).forEach( ( [ dependencyName, version ] ) => {
+				if ( useWorkspace && devDependenciesFilter( dependencyName ) ) {
+					expectedDependencies[ dependencyName ] = PNPM_WORKSPACE_VERSION;
+
+					return;
+				}
```

Comment:
I think we cannot use `devDependenciesFilter()` callback to decide which packages require `workspace:*` version. This callback is designed to work only with dev dependencies, but when we set `useWorkspace: true`, we also want to update normal dependencies as well. It works currently, but the naming does not match the purpose.

Reply:
Updated here [992e410](https://github.com/ckeditor/ckeditor5-dev/pull/1214/commits/992e410fd63581b30387b9970031c8187dcc66cb).

Last thought to consider. Should `workspacePackages` be an array of strings or a callback similar to `devDependenciesFilter`?

It is an array right now, because I went with a simpler implementation.